File: api/NetworkPlanner.Api/NetworkPlanner.Api/Models/FirewallEntry.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json;

namespace NetworkPlanner.Api.Models;

public class FirewallEntry : BaseModel
{
    public HostnameModel Source { get; set; }
    public HostnameModel Destination { get; set; }
    
    public JsonDocument DestinationPorts { get; set; }
    public string Description { get; set; }
    public Environment Environment { get; set; }
    
}

File: shared/models/firewallEntry.ts
import { Environment } from './environment'
import { Hostname } from './hostName'

export interface FirewallEntry {
  id: string
  source: Hostname
  destination: Hostname
  destinationPorts: number[]
  description: string
  environment: Environment
}

export interface FirewallEntryList {
  firewallEntries: FirewallEntry[]
}
